.github/workflows/release-upgradeable.yml

```
@@ -0,0 +1,46 @@
+name: Release Upgradeable
+
+on:
+  workflow_dispatch: {}
+
+jobs:
+  release-upgradeable:
+    environment: push-upgradeable
+    permissions:
+      id-token: write # Required for OIDC
+      contents: read
+    runs-on: ubuntu-latest
+    env:
+      VANILLA_REPO: OpenZeppelin/openzeppelin-contracts
+      UPGRADEABLE_REPO: james-toussaint/openzeppelin-contracts-upgradeable # TODO: Update repo before merging
+    steps:
+      - run: echo "UPGRADEABLE_DIR=${GITHUB_WORKSPACE}/upgradeable" >> "$GITHUB_ENV"
```

I would pass `upgradeable` as a workflow input. Essentially:

```yml
on:
  workflow_dispatch:
    inputs:
          upgradeable-path:
            description: 'The path where the upgradeable repository will be cloned under $GITHUB_WORKSPACE'
            required: true
            type: string
            default: 'upgradeable'
```

So that here we can do:

```suggestion
      - run: echo "UPGRADEABLE_DIR=${GITHUB_WORKSPACE}/${{ github.event.inputs.upgradeable-path }}" >> "$GITHUB_ENV"
```

And:

```yml
- uses: actions/checkout@v5
        with:
          repository: ${{ env.UPGRADEABLE_REPO }}
          submodules: true
          token: ${{ secrets.GH_TOKEN_UPGRADEABLE }}
          ref: ${{ github.ref }}
          path: ${{ github.event.inputs.upgradeable-path }}
```

This way we keep the value consistent across where it's used